Emby.Naming/Video/DateTimeResolvers/DateTimeResolverHelpers.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Emby.Naming.Common;
+
+namespace Emby.Naming.Video.DateTimeResolvers;
+
+/// <summary>
+/// Stuff to reshare.
+/// </summary>
+public class DateTimeResolverHelpers
+{
+    /// <summary>
+    /// Characters that should not be at the start or end of any filename.
+    /// </summary>
+    public static readonly char[] TrimChars = [' ', '-', '.', '_', '*'];
+
+    /// <summary>
+    /// Gets the name of the movie. depending on where the date is found inside the file.
+    /// </summary>
+    /// <param name="fileName">the original name of the file.</param>
+    /// <param name="dateWithSpecialCharacters">what is the date as the full match eg '(2020)'.</param>
+    /// <param name="namingOptions">NamingOptions.</param>
+    /// <returns>The best assumption of the fileName.</returns>
+    public static string GetBestNameMatchAfterRemovalOfDate(string fileName, string dateWithSpecialCharacters, NamingOptions namingOptions)
+    {
+        var positionOfDate = fileName.IndexOf(dateWithSpecialCharacters, StringComparison.InvariantCulture);
+        var dateAtBegginingOfFileName = positionOfDate == 0;
+
+        string withoutDate;
+
+        if (dateAtBegginingOfFileName)
+        {
+            withoutDate = fileName.Substring(dateWithSpecialCharacters.Length);
+        }
+        else
+        {
+            // if we the date is somewhere in the middle of the file we assume the filename is before (by convention)
+            withoutDate = fileName.Substring(0, positionOfDate);
+        }
+
+        // Remove annotation of file
+        if (withoutDate.StartsWith('[') && withoutDate.Contains(']', StringComparison.InvariantCulture))
+        {
+            var withoutPrefix = withoutDate[(withoutDate.IndexOf(']', StringComparison.InvariantCulture) + 1) ..];
```

Comment:
Don't search twice for the closing bracket, use 1 indexof instead and compare the result against -1 to replace the contains

Reply:
adressed in latest commit